docs/src/Button.doc.js

```diff
@@ -12,217 +12,410 @@ const card = c => cards.push(c);
+        defaultValue: null,
+        description: [
+          'Indicate that a button component hides or exposes collapsible components and expose whether they are currently expanded or collapsed.',
+          'Optional with button-role + button-type buttons.',
+          'Accessibility: It populates aria-expanded.',
+        ],
+        href: 'accessibility',
       },
       {
         name: 'accessibilityHaspopup',
         type: 'boolean',
-        description:
-          'Indicates that the element has a popup context menu or sub-level menu. Populates aria-haspopup.',
-        href: 'accessibility-popup',
-      },
-      {
-        name: 'accessibilityLabel',
-        type: 'string',
-        description:
-          'String that clients such as VoiceOver will read to describe the element. Populates aria-label.',
-        href: 'accessibility-popup',
+        required: false,
+        defaultValue: null,
+        description: [
+          'Indicate that a button component controls the appearance of menus and expose whether they are currently opened or closed.',
+          'Optional with button-role + button-type buttons.',
+          'Accessibility: It populates aria-haspopup.',
+        ],
+        href: 'accessibility',
       },
       {
         name: 'color',
-        type: `"blue" | "gray" | "red" | "transparent" | "white"`,
+        type: `'gray' | 'red' | 'blue' | 'transparent' | 'white'`,
+        required: false,
         defaultValue: 'gray',
-        href: 'color',
-      },
-      {
-        name: 'disabled',
-        type: 'boolean',
-        defaultValue: false,
+        description: 'Primary colors to apply to the button background.',
         href: 'combinations',
       },
       {
-        name: 'selected',
+        name: 'disabled',
         type: 'boolean',
-        defaultValue: false,
-        href: 'combinations',
+        required: false,
+        defaultValue: null,
+        description:
+          'Set disabled state so buttons look inactive, cannot be interacted with, and actions are not available.',
+        href: 'type-roles',
       },
       {
         name: 'iconEnd',
         type: '$Keys<typeof icons>',
-        description:
-          'Add an icon to be displayed after the button text. This allows type checking for a valid icon name based on the keys from the list of icons in Icon.',
+        required: false,
+        defaultValue: false,
+        description: [
+          'Add a Gestalt icon to be displayed after the button text. Sometimes an icon can help clarify the usage of a button. Menus are a common use case.',
+          'Accessibility: Icons on buttons are not accessible for screen readers.',
+          'Use IconButton if you only need buttons with icons and no text.',
+        ],
         href: 'iconEnd',
       },
       {
         name: 'inline',
         type: 'boolean',
+        required: false,
         defaultValue: false,
-        href: 'widths',
-      },
-      {
-        name: 'name',
```

Comment:
@AlbertCarreras let's make it explicit that we removed `name` from `Button` by adding a codemod which removes this prop. Even though we might not use it in the Pinterest codebase, we should still be due diligent.

Reply:
@cvuerings, I didn't remove it. Just undocumented.

Reply:
We still have a few instances on Pinboard that we could rename to `id`